crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -449,46 +560,85 @@ async fn get_or_init_server_start_time() -> &'static Instant {
     ONCE.get_or_init(|| async move { Instant::now() }).await
 }
 
+/// Starts an infinite loop that periodically updates the `checkpoint_viewed_at` high watermark.
+pub(crate) async fn update_watermark(
+    db: &Db,
+    checkpoint_viewed_at: Arc<AtomicU64>,
+    metrics: Metrics,
+    sleep_ms: u64,
```

Comment:
nit: can we accept a `Duration` here? Then we don't need to call it `sleep_ms` -- the unit is encoded in the value.